### ISSUE TITLE
Aquamacs Beta 3.6 did not prompt for permissions

### ISSUE THREAD

**Issue by louabill** [aquamacs-emacs/aquamacs-emacs#215]
I downloaded the Aquamacs Beta 3.6 announced by Win on Jun 6, 2021. [1]

When I tried, say, use dired on `~/Documents`, it said it did not have the proper permissions. It did, however, read my `~/.emacs` file, and had no problem getting to files hanging off `~/` but not in `~/Desktop`. After allowing full disk access  and file permissions to my Documents, (and Desktop and Removable Volumes), things were fine.

Aquamacs itself is working fine; it is just that there was no prompting to allow it permissions to do its work.

[1] `Aquamacs 3.6-beta  GNU Emacs 25.3.50.1 (x86_64-apple-darwin20.4.0, NS appkit-2022.44 Version 11.3.1 (Build 20E241))
dated 2021-06-06 rev. 9bcbbdbdd7c9224995d84c009fed9233509c486b`

**Comment by treese** [comment]
@louabill Thanks for the report, and also for trying the beta release. What version of Mac OS are you running? M1 or x86? I'm rather puzzled about this, because I have seen the prompts happen correctly on other systems.

**Comment by treese** [comment]
Thanks for the report. I also received a similar problem on aquamacs-bugs:

> Using iMac (late 2015) and Mac OS 11.4.
> 1. The MacOS menubar seems to require a double-click to activate.

It may be that the changes to Aquamacs to prompt for permissions are insufficient, although they worked in testing, at least on some versions of Mac OS 11.

**Comment by louabill** [comment]
> @louabill Thanks for the report, and also for trying the beta release. What version of Mac OS are you running? M1 or x86? I'm rather puzzled about this, because I have seen the prompts happen correctly on other systems.

I'm on x86.

FWIW, there was some other strangeness caused by the change from Aquamacs 3.5 to the beta. As I tried to run down the problems with the missing permission request, I changed the name of the app from `Aquamacs 3.6 Beta.app` to `Aquamacs.app`, thus clobbering long-set settings associated with Aquamace 3.5. Suddenly applescripts that I use to send text from Emacs to the pasteboard and then to another application failed (and these have worked for many years). As it turned out, the `'ui elements enabled` boolean seems to be permanently `false` and unchangeable via the system preferences.

**Comment by treese** [comment]
A couple of questions:

1. Did you rename the app before or after it failed to prompt for permissions?

2. Where in System Preferences do you have trouble changing "ui elements enabled?

3. In System Preferences->Security & Privacy->Privacy->Automation, is Aquamacs or Aquamacs 3.6 beta enabled for anything?

4. In System Preferences->Security & Privacy->Privacy->Accessibility, is Aquamacs or Aquamacs 3.6 beta enabled?

**Comment by louabill** [comment]
> A couple of questions:
> 
> 1. Did you rename the app before or after it failed to prompt for permissions?

Yes, I did. At first, I zipped up the old `Aquamacs.app` I renamed the beta to `Aquamacs.app`.
After it failed to ask for permissions, AFAICR I then renamed it back to `Aquamacs 3.6 Beta.app`.
I then deleted it, unzipped the old `Aquamacs.app`, started that app, then deleted it, then reinstalled `Aquamacs 3.6 Beta.app` and reran it. I think.

> 2. Where in System Preferences do you have trouble changing "ui elements enabled?

The `ui elements enabled` is a boolean variable built into AppleScript. A long time ago (Mac OS X/OS X/macOS 10.9 and earlier) it was setvia **Enable access for assistive devices**. I'm guessing (but I don't remember) that it was in the  **Universal Access** prefpane.. After that, it was done by checking Aquamacs  under **Accessibility** in the **Security and Privacy** prefpane.

Now, under Big Sur, I am completely clueless about how the `ui elements enabled` gets set. At one point, I played around with checking and unchecking boxes, but stayed `false`. Later, when I had figured out a better plan for testing, it seemed to always be `true`. I feel like a fool.

> 3. In System Preferences->Security & Privacy->Privacy->Automation, is Aquamacs or Aquamacs 3.6 beta enabled for anything?

Yes.

> 4. In System Preferences->Security & Privacy->Privacy->Accessibility, is Aquamacs or Aquamacs 3.6 beta enabled?

Yes.

P.S. Sorry to be so long in replying.

**Comment by treese** [comment]
Thanks agin for the report. 

There's a change to the permissions settings that affects AppleScript in the final version of 3.6, which likely fixes this.

Leaving this open until you can confirm it.

**Comment by louabill** [comment]
I just tried with the 3.6 from December 24, 2021 at 17:07. If this is not the version to which you're referring, please ignore everything below.

The previous behavior is still present: If I shut off all privacy permissions for Aquamacs (they showed in the Accessibility, Full Disk Access, Files and Folders, and Automation items), and then try to run `dired` on my Desktop folder, I get the "Operation not permitted" error, but no prompt for how to fix the problem. I'm guessing that there is still some applescript memory that I've been prompted before. 

Running macOS 12.1 on intel (2017 MBP).

**Comment by treese** [comment]
@louabill Thanks for the update, and for trying 3.6.

Did you quit Aquamacs and restart it for the experiment? Mac OS seems to treat adding or removing permissions while the application is running differently for for the various items.

On an Intel Macbook running 12.1, I tried a fresh installation after deleting all permissions, and it seemed to work as expected.

This leaves me puzzled about what could be wrong, and I haven't thought of a good way to figure it out.
